planner_cspace/src/planner_3d.cpp

```diff
@@ -394,6 +412,129 @@ class Planner3dNode
 
     return true;
   }
+  bool cbMakePlan(nav_msgs::GetPlan::Request &req,
+                  nav_msgs::GetPlan::Response &res)
+  {
+    if (!has_map_)
+      return false;
+
+    if (rough_cost_max_ < 0.01)
```

Comment:
This line checks whether the `fillCostmap()` has already be called or not. If not called, the rough_cost_max_ is near 0.
It would be better the line to be removed.

Reply:
If I remove the line, the planned path might be false best path.
![visualization rviz- - rviz_144](https://user-images.githubusercontent.com/6742467/41956064-54ed3548-7a1d-11e8-9274-cf1a0f433b52.png)

Reply:
Sounds strange.
`fillCostmap` updates distance map from the goal (`cost_estim_cache_`) which is used as a heuristic function in A* algorithm. (This `cost_estim_cache_` is start/goal specific cache; it cannot be used in this service.)

Reply:
Thanks your advice. I've finally found that the problem is the `euclid_cost_coef_` variable that is used in `euclidCost()` has not be properly initialized. The reason why the `fillCostmap` was required is that `euclidCost()` is called in cbCost() for A* algorithm. I've fixed the bug ee30781.

Reply:
ref: final result (after the ee30781)
![visualization rviz- - rviz_145](https://user-images.githubusercontent.com/6742467/41962301-9fa7677c-7a2f-11e8-8fb7-3866b5d7ecae.png)